src/nvme/mi.c

```diff
@@ -812,6 +865,38 @@ int nvme_mi_admin_identify_partial(nvme_mi_ctrl_t ctrl,
 	return 0;
 }
 
+int nvme_mi_service_primitive_control(nvme_mi_ep_t ep, __u8 opcode, __u16 *result_cpsr)
+{
+	struct nvme_mi_service_resp_hdr resp_hdr;
+	struct nvme_mi_service_req_hdr req_hdr;
+	struct nvme_mi_resp resp;
+	struct nvme_mi_req req;
+	int rc = 0;
+	__le16 cpsr = 0xffff;
+
+	nvme_mi_service_init_req(&req, &req_hdr, opcode);
+
+	nvme_mi_calc_req_mic(&req);
```

Comment:
The MIC should be calculated by `nvme_mi_submit()`, you don't need this.

Reply:
yup, remove it